CRM/Report/Form.php

```diff
@@ -4893,4 +4893,23 @@ public function joinGroupTempTable($baseTable, $field, $tableAlias) {
     }
   }
 
+  /**
+   * Get all labels for credit card type rows in report.
+   *
+   * @param string $creditCardTypes
+   *   comma separated option values of option group "credit_card_type".
+   */
+  public function getGroupCreditCardType($creditCardTypes) {
+    $cardTypes = CRM_Core_PseudoConstant::get('CRM_Financial_DAO_FinancialTrxn', 'card_type');
+    $types = explode(',', $creditCardTypes);
+    $labels = array();
+    foreach ($types as $value) {
+      $label = CRM_Utils_Array::value($value, $cardTypes);
+      if (!in_array($label, $labels)) {
+        $labels[] = $label;
+      }
+    }
+    return implode(', ', array_filter($labels));
+  }
+
```

Comment:
I agree this is an overly specific function to meet the general need of matching a pseudoconstant to its values. I would be very surprised if there isn't already a generic way of handling this in the report class. What do other columns use to format their output?